lib/tasks/akismet.rake

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+desc 'Checks old posts using Akismet'
+task 'akismet:scan_old' => :environment do
+  # Scan the first post of users that match:
+  # - regular posts only
+  # - not deleted posts
+  # - post contains a URL
+  # - user is not trusted (TL0-2 and not staff)
+  # - no other post of this user was checked before
+  sql = <<~SQL
+    WITH first_user_posts AS (
+      SELECT MIN(posts.id) post_id, posts.user_id
+      FROM posts
+      JOIN topics on topics.id = posts.topic_id
+      WHERE (raw LIKE '%http%' OR cooked LIKE '%href%')
```

Comment:
I am not so sure about this condition. Other option would be to look for `://` in raw or cooked?

Reply:
We do autolink URLs like this `google.com` so `://` won't catch them. I think your conditions are fine 👍

Reply:
`google.com` becomes `<a href="http://google.com">google.com</a>`, so :// would still catch those. In this case I am thinking of getting rid of `raw LIKE '%http%'` because it will not work with autolink.

Reply:
Yeah, whatever is fine. There's no silver bullet other than re-parsing the raw/cooked to detect URL...